<launch>
    <arg name="marker_size" value="0.07" doc="Size of the ArUco marker used, in meters" />
    <arg name="marker_id" value="100" doc="The ID of the ArUco marker used" />

    <!-- start ArUco -->
    <node name="aruco_tracker" pkg="aruco_ros" type="single">
        <remap from="/camera_info" to="/camera/color/camera_info" />
        <remap from="/image" to="/camera/color/image_rect_color" />
        <param name="image_is_rectified" value="True"/>
        <param name="marker_size"        value="$(arg marker_size)"/>
        <param name="marker_id"          value="$(arg marker_id)"/>
        <param name="reference_frame"    value="camera_link"/>
        <param name="camera_frame"       value="camera_color_optical_frame"/>
        <param name="marker_frame"       value="camera_marker" />
        <!-- <param name="reference_frame"    value="camera_link"/>
        <param name="camera_frame"       value="camera_color_optical_frame"/> -->
    </node>

    <!-- start easy_handeye -->
    <include file="$(find easy_handeye)/launch/calibrate.launch" >
        <arg name="eye_on_hand" value="true" />
        <arg name="freehand_robot_movement" value="true"/>
        <arg name="tracking_base_frame" value="camera_link" />
        <arg name="tracking_marker_frame" value="camera_marker" />
        <arg name="robot_base_frame" value="panda_link0" />
        <arg name="robot_effector_frame" value="panda_EE" />
        <arg name="start_rviz" value="false"/>
    </include>
</launch>